src/borg/item.pyx

```diff
@@ -359,60 +377,127 @@ class ManifestItem(PropDict):
+
+    def __repr__(self):
+        if self.equal:
+            return 'equal'
+
+        changes = []
+
+        if self.item1.is_link() or self.item2.is_link():
+            changes.append(self._link_string())
+
+        if 'chunks' in self.item1 and 'chunks' in self.item2:
+            changes.append(self._content_string())
+
+        if self.item1.is_dir() or self.item2.is_dir():
+            changes.append(self._dir_string())
+
+        if not (self.item1.get('deleted') or self.item2.get('deleted')):
+            changes.append(self._owner_string())
+            changes.append(self._mode_string())
+
+        return ' '.join((x for x in changes if x))
+
+    def _equal(self, chunk_iterator1, chunk_iterator2, can_compare_chunk_ids):
+        # if both are deleted, there is nothing at path regardless of what was deleted
+        if self.item1.get('deleted') and self.item2.get('deleted'):
+            return True
+
+        attr_list = ['deleted', 'mode', 'source']
+        if self._numeric_owner:
+            attr_list += ['uid', 'gid']
+        else:
+            attr_list += ['user', 'group']
+        for attr in attr_list:
+            if self.item1.get(attr) != self.item2.get(attr):
+                return False
+
+        if 'mode' in self.item1:     # mode of item1 and item2 is equal
+            filemode = stat.filemode(self.item1.mode)
+            if filemode[0] == 'l' and 'source' in self.item1 and 'source' in self.item2 and self.item1.source != self.item2.source:
+                return False
+
+        if 'chunks' in self.item1 and 'chunks' in self.item2:
+            return self._content_equal(chunk_iterator1, chunk_iterator2, can_compare_chunk_ids)
+
+        return True
+
+    def _link_string(self):
+        if self.item1.get('deleted'):
             return 'added link'
-        if item2.get('deleted'):
+        if self.item2.get('deleted'):
             return 'removed link'
-        if 'source' in item1 and 'source' in item2 and item1.source != item2.source:
+        if 'source' in self.item1 and 'source' in self.item2 and self.item1.source != self.item2.source:
             return 'changed link'
-
-def compare_content(path, item1, item2):
-    if contents_changed(item1, item2):
-        if item1.get('deleted'):
-            return ('added {:>13}'.format(format_file_size(sum_chunk_size(item2))))
-        if item2.get('deleted'):
-            return ('removed {:>11}'.format(format_file_size(sum_chunk_size(item1))))
-        chunk_ids1 = {c.id for c in item1.chunks}
-        chunk_ids2 = {c.id for c in item2.chunks}
+        return ''
+
+    def _content_string(self):
+        if self.item1.get('deleted'):
+            return ('added {:>13}'.format(format_file_size(self.item2.get_size())))
+        if self.item2.get('deleted'):
+            return ('removed {:>11}'.format(format_file_size(self.item1.get_size())))
+        # How is working with ids correct if we cannot compare ids (i.e. can_compare_chunk_ids == False)?
```

Comment:
When comparing chunks, the length of a memoryview of a chunk is its size in bytes, right?
I could then fix this, but probably better in a separate PR, as it is unrelated.